File: ruby/gmsec4/dox/TimeUtil.rb
# Copyright 2007-2021 United States Government as represented by the
# Administrator of The National Aeronautics and Space Administration.
# No copyright is claimed in the United States under Title 17, U.S. Code.
# All Rights Reserved.

# This file is used for generating Ruby documentation; it only contains
# stubs of constructors and methods associated with the class.


module Libgmsec_ruby #:nodoc: don't document this

	# This class defines the logging interface.
	#
	class TimeUtil
		##
		# Sleeps for the specified number of milliseconds.
		#
		# ==== Attributes
		#
		# * +time_in_ms+ - The period (in milliseconds) to sleep; minimum allowed period is 1 ms.
		#
		def self.millisleep(time_in_ms)
		end


		##
		# Returns the current time in seconds since Jan 1, 1970.
		#
		# ==== Returns
		#
		# A GMSEC_TimeSpec object is returned.
		#
		def self.get_current_time
		end


		##
		# Returns the current time in seconds since Jan 1, 1970 in the format of a floating point number.
		#
		# If the given time_spec object is non-nil, it will be populated with the current time.
		#
		# ==== Attributes
		#
		# * +time_spec+ - Optional GMSEC_TimeSpec object
		#
		# ==== Returns
		#
		# A float point number is returned.
		#
		def self.get_current_time_s(time_spec = nil)
		end


		##
		# Returns a GMSEC_TimeSpec object that is derived from the given time string.
		#
		# The time string must have a format similar to YYYY-DDD-HH:MM:SS[.s*]
		#
		# ==== Attributes
		#
		# * +time_str+ - The time string.
		#
		# ==== Returns
		#
		# A GMSEC_TimeSpec object is returned.
		#
		def self.get_char_time(time_str)
		end


		##
		# Returns time, derived from the given time string, since Jan 1, 1970, as a floating point value.
		#
		# The time string must have a format similar to YYYY-DDD-HH:MM:SS[.fffffffff]
		#
		# ==== Attributes
		#
		# * +time_str+ - The time string.
		#
		# ==== Returns
		#
		# A floating point time value is returned.
		#
		def self.get_char_time_s(time_str)
		end


		##
		# Returns a GMSEC-style time string.
		#
		# The time string has a format of YYYY-DDD-HH:MM:SS[.s*]. The given GMSEC_TimeSpec, when interpreted as an absolute time value,
		# represents the number of seconds elapsed since Jan 1, 1970.
		#
		# ==== Attributes
		#
		# * +time_spec+ - A GMSEC_TimeSpec object containing time in seconds and nanoseconds.
		# * +subs+      - Optional parameter to indicate the number of digits (0 through 9) to allow for subseconds; default is 3.
		#
		# ==== Returns
		#
		# A time string.
		#
		# ==== Example usage
		#
		#    current_time = Libgmsec_ruby::TimeUtil::format_time( Libgmsec_ruby::TimeUtil::get_current_time )
		#
		def self.format_time(time_spec, subs = 3)
		end


		##
		# Computes the time-delta between the two given times (t1 - t0).
		#
		# ==== Attributes
		#
		# * +t1+ - The second/later GMSEC_TimeSpec object.
		# * +t0+ - The first/earlier GMSEC_TimeSpec object.
		#
		# ==== Returns
		#
		# A GMSEC_TimeSpec object.
		#
		# ==== Example usage
		#
		#    t0_time = Libgmsec_ruby::TimeUtil::get_current_time
		#    # ... some delay
		#    t1_time = Libgmsec_ruby::TimeUtil::get_current_time
		#
		#    delta_time = Libgmsec_ruby::TimeUtil::diff( t1_time, t0_time )
		#
		def self.diff_time(t1, t0)
		end
	end
end

File: csharp/gmsec4/dox/util/TimeUtil.cs
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

/* This file is used for generating Doxygen documentation; it only contains
   stubs of constructors and methods associated with the class.
*/


namespace GMSEC.API {

/// <summary>
/// Contains the methods that support the acquisition of current
/// system time, and the ability to format such into a GMSEC-style time
/// string.
/// </summary>

public class TimeUtil : global::System.IDisposable {

  /// <summary>Sleeps for the specified number of milliseconds (at least 1).</summary>
  ///
  /// <param name="milliseconds">The amount of time in milliseconds to sleep</param>
  public static void Millisleep(uint milliseconds) {
  }

  /// <summary>Returns the current time in seconds since Jan 1, 1970.</summary>
  public static GMSEC_TimeSpec GetCurrentTime() {
  }

  /// <summary>
  ///  Returns the current time in (real) seconds since Jan 1, 1970.
  /// <p>
  /// If the GMSEC_TimeSpec is non-null, it will be populated.
  /// </summary>
  ///
  /// <param name="ts">The GMSEC_TimeSpec object, if any, to populate</param>
  ///
  /// <returns>The current time as a double value</returns>
  public static double GetCurrentTime_s(GMSEC_TimeSpec ts) {
  }

  /// <summary>Returns the current time in (real) seconds since Jan 1, 1970.</summary>
  ///
  /// <returns>The current time as a double value</returns>
  public static double GetCurrentTime_s() {
  }

  /// <summary>Returns the current time in seconds since Jan 1, 1970 as represented by the %GMSEC String.</summary>
  ///
  /// <param name="timeString">The GMSEC-style time string to convert</param>
  ///
  /// <returns>A GMSEC_TimeSpec object that contains the converted time</returns>
  public static GMSEC_TimeSpec GetCharTime(string timeString) {
  }

  /// <summary>Returns the current time in seconds since Jan 1, 1970 as represented by the %GMSEC String.</summary>
  ///
  /// <param name="timeString">The GMSEC-style time string to convert</param>
  ///
  /// <returns>The time as a double value</returns>
  public static double GetCharTime_s(string timeString) {
  }

  /// <summary>Computes the time-delta between the two given times.</summary>
  ///
  /// <param name="time1">The end time</param>
  /// <param name="time0">The start time</param>
  ///
  /// <returns>A GMSEC_TimeSpec object that contains the time-delta between the two given times</returns>
  public static GMSEC_TimeSpec DiffTime(GMSEC_TimeSpec time1, GMSEC_TimeSpec time0) {
  }

  /// <summary>Formats the given GMSEC_TimeSpec into a GMSEC-style time string with the format of YYYY-DDD-HH-MM-SS.sss</summary>
  ///
  /// <param name="ts">The GMSEC_TimeSpec containing the time to represent as a string</param>
  ///
  /// <returns>A GMSEC-style formatted time string</returns>
  public static string FormatTime(GMSEC_TimeSpec ts) {
  }

  /// <summary>Formats the given GMSEC_TimeSpec into a GMSEC-style time string with the format of YYYY-DDD-HH-MM-SS[.s*]</summary>
  ///
  /// <param name="ts">The GMSEC_TimeSpec containing the time to represent as a string</param>
  /// <param name="subs">The number of digits to allow for subseconds; range is 0 through 9.</param>
  ///
  /// <returns>A GMSEC-style formatted time string</returns>
  public static string FormatTime(GMSEC_TimeSpec ts, int subs) {
  }

  private TimeUtil() {
  }

}

}
